app/lib/clients/vault.rb

```diff
@@ -25,23 +28,89 @@ def kv_delete(path)
 
       def client
         ::Vault::Client.new(
-          address: Rails.configuration.astral[:vault_addr],
+          address: vault_address,
           token: Rails.configuration.astral[:vault_token]
         )
       end
 
+      def vault_address
+        Rails.configuration.astral[:vault_addr]
+      end
+
+      def kv_mount
+        "kv_astral"
+      end
+
+      def intermediate_ca_mount
+        "pki_astral"
+      end
+
+      def cert_path
+        "#{intermediate_ca_mount}/issue/astral"
+      end
+
+      def root_ca_ref
+        Rails.configuration.astral[:vault_root_ca_ref]
+      end
+
+      def root_ca_mount
+        Rails.configuration.astral[:vault_root_ca_mount]
+      end
+
       def enable_engine(mount, type)
-        # create the engine mount if not present already
-        unless client.sys.mounts.key?(mount.to_sym)
-          client.sys.mount(mount, type, "#{type} secrets engine")
-        end
-      rescue ::Vault::HTTPError => e
-        Rails.logger.error "Error enabling #{type} engine: #{e}"
+        client.sys.mount(mount, type, "#{type} secrets engine")
       end
 
+      def enable_ca
+        # if mount exists, assume configuration is done
+        if client.sys.mounts.key?(intermediate_ca_mount.to_sym)
+          return
+        end
 
-      def kv_mount
-        Rails.configuration.astral[:vault_kv_mount]
+        # create the mount
```

Comment:
Can we test this in some way? It does a bunch of stuff here!

Reply:
it's implicitly tested by the integration tests but I'll work on adding more

Reply:
tests added for the 'configure_pki' and 'configure_kv'